src/ip_data.py

```diff
@@ -105,46 +105,61 @@ def init_tables(self) -> None:
         pass
 
     @abstractmethod
-    def __enter__(self):
-        return self
-
-    @abstractmethod
-    def __exit__(self, exc_type, exc_val, exc_tb):
+    def close(self) -> None:
         pass
 
 
 class JSONFileWriter(FileWriter):
-    def _write_to_json(self, filename: str, schema: List[str], generator: Iterator[List[Any]]) -> None:
-        result = []
+    def __init__(self, base_filename: str):
+        super().__init__(base_filename)
```

Comment:
the base class is `FileWriter`, which is an abstract base class, this is not required

Reply:
removed